### ISSUE TITLE
Build new site showing awesome-selfhosted data

### ISSUE THREAD

**Issue by jamesread** [awesome-selfhosted/awesome-selfhosted-data#912]
Hey there, I tried reaching out via email a day or two ago, but I'd like to also ask publicity; I have a new idea for how to display the awesome-selfhosted data in a way that some members of the community might find useful. 

I'd like to not share the concept and site here, but discuss privately simply because I don't want to rip off the amazing work of this repo & contributors without your go ahead. My ideal outcome would be to contribute the idea back to this repo / organization - I have no commercial intents or anything like that. 

My email address is contact@jread.com - can I share you some screenshots and links privately, and you say if you want me to contribute back, or stop, or go ahead but do my own thing? Thanks!

**Comment by nodiscc** [comment]
Hi @jamesread sorry for not taking the time to reply to your email yet, let's continue this discussion here - openness and transparency have characterized this project since the beginning.


Can you tell us more about this proposed way to display the data? (I already know for your email but again, let's discuss this publicly - it seemed interesting).

Anyone is free to reuse our data in any way they want, it is licensed under [CC-BY-SA-3.0](https://creativecommons.org/licenses/by-sa/3.0/) which only requires that you give appropriate credit by linking back here in a visible way, and that your derived work stays under the same license.

About possible collaborations, we could implement this in any way we find comfortable, for example:
- you maintain the tooling to update/generate the proposed website, host it on your own servers
- we maintain the tooling in collaboration in a repository under the https://github.com/awesome-selfhosted/ organization, you host it on your own server
- same as above, but we host it on the the server already hosting https://awesome-selfhosted.net/, under a subdomain. In this case, the deployment must be automated (as is the case with the current [ansible role](https://gitlab.com/nodiscc/toolbox/-/tree/master/ARCHIVE/ANSIBLE-COLLECTION/roles/awesome_selfhosted_html) that deploys and updates awesome-selfhosted.net automatically. In fact it is expected that any tools used would be Free and Open-Source)

In any case, the main constraint is _time_ required to maintain the solution - I barely have time to maintain the current list + the [tooling](https://github.com/nodiscc/hecat/issues?q=is%3Aopen+) responsible for automated maintenance/website generation), as all people involved are volunteers. So a high level of autonomy would be expected.

**Comment by jamesread** [comment]
Hey @nodiscc , thanks so much for taking the time to reply - and I also appreciate that we can have this discussion in the open too, brilliant ;-) 

So I as a huge fan of awesome-selfhosted myself (I even have OliveTin in the list!), one thing I found myself wanting to do was to work out almost a selfhosted "roadmap" for myself, or visualize the landscape of available options in an ecosystem. For me, I really like the visual offered by the CNCF landscape - http://l.cncf.io (which is very much aimed at the kubernetes community). 

From knowing that the CNCF landscape software itself is open source, I just span up a quick experiment on my machine, a python script to injest the .yml files of the selfhosted data, and spit it back out again in a monolithic .yaml file needed by the CNCF landscape software. I then added the dashboard-icons repo, and the screenshots below are what I ended up with - talking about standing on the shoulders of 3 great projects, I hacked this together in an hour or so! 

![image](https://github.com/user-attachments/assets/ff589887-c4ee-4946-8958-8d1b6806654a)

![image](https://github.com/user-attachments/assets/9ce26c63-9a78-4892-9d77-3384652f04e4)

What I feel this format gives me over and above the existing html format at https://awesome-selfhosted.net/ is the came categories, that you have, just with the simple project icons which I find less visually overloading to browse. 

Where I'd like to go from here, is to add the ability to mark projects as "todo", "not interested", and "have this in my stack". This would make it really nice for people to plan out their own self hosted roadmaps for themselves, and explore the interesting projects in this list. 

I dare say it, but this site could easily be a docker container, which people can self host, if they want to keep their selfhosted "roadmaps" private. Equally, it would be nice to filter "just the stuff I'm running" for screenshots :-) 

I'm quite sure there are many people that value the information it's current format, so I'm not suggesting a replacement of awesome-selfhosted.net - my proposal here is for a new representation of the same data, just visually, with a mechanism to tag/edit the data for their personal needs. 

I'd be more than happy to host myself, host on awesome-selfhosted.net under a subdomain (optionally on other servers), or just create a new project like "my-selfhosted-roadmap" that people just self host themslves, using the data? 

My biggest fear about hosting an online version is the sheer bandwidth consumed by all the icons! There are certainly optimizations available there (like making it a couple of large imagemaps), however.

**Comment by nodiscc** [comment]
I am familiar with https://landscape.cncf.io/ (also involved in https://github.com/awesome-foss/awesome-sysadmin and we delegated part of the software catalog to the CNCF landscape since it is much more extensive, and the various views/options are very useful - I happen to prefer the "Card" view :) )

I think this would definitely be an interesting alternative to the existing site.

> My biggest fear about hosting an online version is the sheer bandwidth consumed by all the icons

* There are 2535 entries in the CNCF landscape (`curl -s https://raw.githubusercontent.com/cncf/landscape/refs/heads/master/landscape.yml | grep -c '          name:'`)
* and 1220 in awesome-selfhosted-data (`ls awesome-selfhosted-data/software/ | wc -l`).
* https://landscape.cncf.io/ loads 18MB of svg files
* This would put us at ~8MB per page load (only the first page load, these will be cached for a few hours afterwards).
* There are ~2000 daily unique visitors on https://awesome-selfhosted.net/ (I wrote somewhere that I would share the stats but didn't have time yet)
* which would put us at... 17GB bandwidth per day :rofl: if the new site receives the same amount of traffic.
* Moving the icons to a large image map would only reduce the number of requests (actually not since we use HTTP2), not the total file size.
* I think the hosting provider could handle this without problems though, even if it kind of feels like a waste.

> a python script to injest the .yml files of the selfhosted data, and spit it back out again in a monolithic .yaml file needed by the CNCF landscape software.

This is great, are you willing to publish this tool under a FOSS license? We could arrange the organizational/ownership/maintenance model stuff at a later time.

> add the ability to mark projects as "todo", "not interested", and "have this in my stack". This would make it really nice for people to plan out their own self hosted roadmaps for themselves

Agreed, wouldn't this require maintaining a fork of the CNCF landscape software though?

> this site could easily be a docker container, which people can self host, if they want to keep their selfhosted "roadmaps" private

Also agreed, and regardless of the hosting/who-does-what option we pick later, having the possibility to generate the site locally is a requirement anyway.

> OliveTin

:ok_hand: tried it briefly a while ago, and know it's in my list of things to add to my stack. I use Gitea issues to manage this "todo list" but the proposed tool would also work :)

**Comment by majicmaj** [comment]
Hi everyone,

I’d love to help build on the awesome-selfhosted project! I’ve put together a working concept that reimagines the data display while staying true to the project’s open, FOSS ethos. You can check out the live demo and source code here:

- **Demo:** [[awesomeselfhosted.netlify.app](https://awesomeselfhosted.netlify.app/)](https://awesomeselfhosted.netlify.app/)
- **GitHub:** [[majicmaj/awesome-selfhosted-ui](https://github.com/majicmaj/awesome-selfhosted-ui)](https://github.com/majicmaj/awesome-selfhosted-ui)

**Key features so far:**

- **Catalog Page:** A clean, responsive view of the full project list.  
  <img width="1512" alt="Catalog page" src="https://github.com/user-attachments/assets/de115fb0-fafc-4a11-9c17-2a3f0ee72e95" />

- **Detailed Project View:** Each project has its own detailed view for better insight.  
  <img width="1512" alt="Detailed view" src="https://github.com/user-attachments/assets/9c277824-fa28-4830-b33a-129d35d190d6" />

- **New Projects Spotlight:** Quickly spot newly added projects.  
  <img width="1512" alt="New projects" src="https://github.com/user-attachments/assets/8b5096a7-d1b3-47e2-a323-151239223fda" />

- **Discovery Mode:** Explore a randomized selection from various categories for inspiration.  
  <img width="1512" alt="Discovery mode" src="https://github.com/user-attachments/assets/aa6885f9-825e-49ae-a829-e3a2adeb9425" />

I love the idea of allowing users build their own self-hosted roadmaps, I'd be happy to tackle that as well. I’m happy to work with you on whichever hosting or integration approach makes the most sense—whether that’s hosting on my own servers, collaborating under the awesome-selfhosted organization, or integrating as a subdomain.

One quick note regarding licensing: I’m currently using the MIT license for my code. Given that the awesome-selfhosted data is under CC-BY-SA-3.0, I’d love to know if this is acceptable as long as proper attribution is maintained, or if you’d prefer the UI project to adopt a similar license. I’m open to adjusting this to best fit the project’s requirements.

Looking forward to your feedback and hoping we can collaborate on this exciting enhancement for the community!

**Comment by nodiscc** [comment]
Hi @majicmaj, thanks for submitting this here. The demo looks great.

Here is what we could do:
- I try to build the site locally, check that everything works as expected
- I deploy it on the server currently hosting https://awesome-selfhosted.net/ under a separate vhost (e.g. `ui.awesome-selfhosted.net`, we can think about the best subdomain name later), of course I would make the deployment tooling available under a FOSS license as is the case with our [current deployment tolling](https://github.com/nodiscc/toolbox/tree/master/ARCHIVE/ANSIBLE-COLLECTION/roles/awesome_selfhosted_html)
- we find the best way to handle updates
- (optional) we move your repository under the awesome-selfhosted organization (you keep ownership of the repo, of course)
- we add a link to the new site in the current website/markdown list header
- we keep improving it over time, when we think it's ready we make an "official" announcement
- ...

>  I’m currently using the MIT license for my code. Given that the awesome-selfhosted data is under CC-BY-SA-3.0,

MIT is fine, [AGPL-3.0](https://choosealicense.com/licenses/agpl-3.0/) would be even better. CreativeCommons licenses are fine for datasets but not ideal for software projects.

What do you think?

One question, how are we supposed to update the dataset used by your app? Does it pull the data "live" from somewhere at runtime?

**Comment by majicmaj** [comment]
@nodiscc That sounds good to me! Also would be happy to adjust to using AGPL-3.0

> One question, how are we supposed to update the dataset used by your app? Does it pull the data "live" from somewhere at runtime?

Currently, it parses the readme from github on the client side on the first load to fetch the latest data and it's cached locally (24hrs default, adjustable in settings) 

I'd like to move to generating the data ahead such as on build time or being able to fetch a computed version of the list in JSON instead of the markdown file.

What would be the best way for us to discuss this further? If you use discord, mine is @monkus.aurelius

**Comment by Rabenherz112** [comment]
> CNCF landscape

Today, I came somehow back back to this issue after being intrigued by the CNCF landscape visualisation, and hacked together a quick Python proof‑of‑concept (still missing features like redirects and automatic icon fetching). It pulls in the latest awesome‑selfhosted‑data repo and spins up a CNCF‑style landscape website.

So far, I’m pretty happy with how it looks, but I’ve run into a few limitations of the landscape generator (natively):

- Required subcategories & descriptions
Every top‑level category must have at least one subcategory and its own description. Since our dataset doesn’t define subcategories for some areas (e.g. Analytics), it forces these odd empty placeholders for the "Guide" page.
https://fs.ravshort.com/XhYQq.png

- GitHub API rate limits
To get just any metadata like licenses or star counts, you need the built‑in GitHub API integration, which rapidly burns through rate limits. While we have this data (partially already in our repo), I haven’t found a way to display that info without relying on this feature.

- No custom tags
I also am missing custom tags for the programming language for example. While I was able to add these tags to the dataset of cncf itself, it just won't display that (though to be fair, this could also just be me, not completely understanding it)

If anybody is curious to see it in action (minus GitHub metadata, because that just won't work for the full repo -_-), check out the temporary demo:
https://demo-as-landscape.theravenhub.com

**Comment by Rabenherz112** [comment]
Heya, it's me again :D

**TL;DR**
I've built an alternative static website to present our dataset and would appreciate if at least the main reviewers (i.e @nodiscc , @Ki-er , and @kokomo123 ) could take some time to explore it. Though of course anybody seeing this, is more then welcome to give their feedback on this.

https://awesome-web.theravenhub.com/
(this version is automatically updated every day) 

**Teeeexxt**
To contribute to our ongoing discussion, I decided to experiment with a custom frontend that showcases our data in a new light. Although I don't expect this prototype to replace our current solution, mainly because it relies on client‑side JavaScript for rendering the browse page and will simply fail for users who disable scripts as well as the fact, the it is not yet really mature. I do believe it could serve as a valuable alternative view, perhaps as a subdomain of awesome-selfhosted? (open to suggestions here)

I built this as a static website generator using a Python script so that it stays as close as possible to our existing technology stack. The biggest challenge was implementing the dynamic filtering and sorting features; I explored several approaches but ultimately settled on rendering the browse page in the browser using JavaScript to deliver the fluid experience users expect. On the plus side, this site also automatically generates commit‑history graphs to visualize each project's activity over time and parses special description flags such as "fork of", "alternative to", or "documentation only in", displaying them with distinct styling.

Since I'm coming from a sysadmin background rather than as a web developer, I leaned on AI/LLMs to help write much of the code. If that's a concern, I completely understand. In any case, all of the generator's source code is open source under the AGPL‑v3 license at https://github.com/Rabenherz112/awesome-selfhosted-web-gen/, and you can find the current configuration for the automatically updated site at https://github.com/Rabenherz112/awesome-selfhosted-web. I'd love to hear your thoughts and any feedback you have on this version!

**Comment by kokomo123** [comment]
This is definitely a significant improvement over the original website, I like how you can filter through categories, the icons are better (the addition of the special description flags), however, I think the new feature set website with the detailed statistics is in a weird spot while scrolling down, I definitely think we can keep it in its own separate tab if people like, or rearrange it so its more clear where it is. I don't have any notes besides that feedback. I would also like to note since you used a significant amount of AI/LLMs, to probably write down the models you used for transparency. AI/LLMs for me I am still torn about, considering the environmental costs you use with them depending on which one, though you can self host them which is a bonus. The code I am not an expert on so i'll let others weigh in if we should use a significant chunk be AI/LLM code or rewrite it to help us a little bit rather then a huge chunk.

**Comment by Rabenherz112** [comment]
Thanks for the review kokomo.
I added as suggested a Disclaimer regarding the use of AI/LLMs.

What did you mean with
>  I think the new feature set website with the detailed statistics is in a weird spot while scrolling down

Couldn't figure it out, sorry.

**Comment by kokomo123** [comment]
<img width="1366" height="405" alt="Image" src="https://github.com/user-attachments/assets/e75fa010-9a7d-47e4-a8ac-c851b13417fa" />
Ah, sorry for not elaborating on that note. I meant when scrolling down, seeing this 5 scrolls down instead of like at the top or whatever is weird in my own opinion, others may consider it fine i guess, im not sure. Also I mean, it could just stay in the detailed statistics tab or stay where it is currently.

**Comment by Rabenherz112** [comment]
Ah that is what you meant...
This is configurable. You can disable this section if wanted in the config, in which case it is just not displayed on the homepage.

<img width="380" height="226" alt="Image" src="https://github.com/user-attachments/assets/12d37150-972e-4c36-9ba7-bafd91042f42" />

**Comment by kokomo123** [comment]
Alright, no issues here then. 👍🏻

**Comment by jamesread** [comment]
Hey all, 

It's great to see now 3 proposed different UIs in this thread (my original suggestion with the CNCF landscape app, @majicmaj 's very nice looking "ASH UI", and @Rabenherz112 's very nice looking "ravenhub UI"). I'm not possessive about which solution "wins", of course, just that there seem to be a few of us with similar ideas, and merging efforts is always good! 

I must admit - I chose a bad issue title when raising this issue - simply "**build a new site**"... it doesn't actually describe what I wanted to do. Actually, the title I should have used is something like "**make it easier for people to plan their selfhosted roadmap**". I explained this in my original 2nd post in this thread, but let me copy it here again to clarify - I said this;

> Where I'd like to go from here, is to add the ability to mark projects as "todo", "not interested", and "have this in my stack". This would make it really nice for people to plan out their own self hosted roadmaps for themselves, and explore the interesting projects in this list.

At the same time, rightfully concerns have been raised about how to efficiently serve project icons - this is certainly doable, but I'd be really interested in avoiding that problem (at first) by making the project itself self-hostable. 

I'm thinking about how we could collaboratively and collectively move forward - may I suggest the following - how do @Rabenherz112 , and @majicmaj feel about; 

* Merging the above suggestions into 1 codebase we share / contribute to? It certainly looks like "ASH UI" has a great following of GitHub stars, and personally I like the interface a little more. Would there be any objection about making this the UI that we collaborate on? 
* Extending the functionality **to allow a self-hoster to tag projects** - the tags I'd want to define for myself would be "not interested", "todo", "using", or something like that. 

If we think this makes sense, I'd raise a feature request in the ASH UI project and try and put some of my energy there!

**Comment by Rabenherz112** [comment]
Sorry for my late response here, I was on vacation :D.

Merging efforts is always the right way forward, so I have no objection to this approach. Since I also don't have a strong opinion on which solution should be developed further / used, I’m fine with continuing work on ASH UI by majicmaj.

That said, I wouldn’t personally be able to contribute or maintain much to the codebase itself, as I've never really worked with TypeScript (I tried it once, failed miserably and went back to JavaScript).

Edit:
I also quite like the Idea with the tagging of projects. Implementing this in my project would probably be somewhat difficult, since it is a static webpage.
